supertokens_python/recipe/emailverification/api/implementation.py

```diff
@@ -13,65 +13,127 @@
+    async def is_email_verified_get(self, api_options: APIOptions, user_context: Dict[str, Any]) -> IsEmailVerifiedGetOkResponse:
+        """
+        is_email_verified_get is an async method that accepts
+
+            - api_options: APIOptions
+            - user_context: Dict
+
+        returns a json/dict response as follows if the email is verified
+
+            {
+                "status": "OK",
+                "isVerified": true
+            }
+
+        Otherwise it raises an exception.
+        """
+
+        # TODO: fix docstring, docstring should mention exact functioning of this method.
+        # TODO: this method should return an error response instead of an exception.
+
         session = await get_session(api_options.request)
-        if session is None:
-            raise Exception('Session is undefined. Should not come here.')
 
-        user_id = session.get_user_id(user_context)
-        email = await api_options.config.get_email_for_user_id(user_id, user_context)
+        if isinstance(session, SessionContainer):
+            user_id = session.get_user_id(user_context)
+            email = await api_options.config.get_email_for_user_id(user_id, user_context)
+            is_verified = await api_options.recipe_implementation.is_email_verified(user_id, email, user_context)
+            return IsEmailVerifiedGetOkResponse(is_verified)
 
-        is_verified = await api_options.recipe_implementation.is_email_verified(user_id, email, user_context)
-        return IsEmailVerifiedGetOkResponse(is_verified)
+        raise Exception("Undefined Session")
 
     async def generate_email_verify_token_post(self, api_options: APIOptions, user_context: Dict[str, Any]) -> GenerateEmailVerifyTokenPostResponse:
+        """
+        generate_email_verify_token_post is an async method that accepts
+
+            - api_options: APIOptions
+            - user_context: Dict
+
+        returns a json/dict response as follows
+
+        In the case of success:
+
+            {
+                "status": "OK"
+            }
+
+        In the case of failure:
+            {
+                "status": "EMAIL_ALREADY_VERIFIED_ERROR"
+            }
+        """
         session = await get_session(api_options.request)
-        if session is None:
-            raise Exception('Session is undefined. Should not come here.')
+        if isinstance(session, SessionContainer):
+            user_id = session.get_user_id(user_context)
+            email = await api_options.config.get_email_for_user_id(user_id, user_context)
 
-        user_id = session.get_user_id(user_context)
-        email = await api_options.config.get_email_for_user_id(user_id, user_context)
+            token_result = await api_options.recipe_implementation.create_email_verification_token(user_id, email, user_context)
+            if token_result.is_email_already_verified:
```

Comment:
You should use isinstance to check the response type from that function call - as discussed earlier.